client/components/domains/contact-details-form-fields/managed-contact-details-form-fields.jsx

```diff
@@ -321,12 +323,24 @@ export class ManagedContactDetailsFormFields extends Component {
 	}
 
 	renderAlternateEmailFieldForGSuite() {
+		let customErrorMessage = this.props.contactDetailsErrors?.alternateEmail;
+		// We also show 'email' field errors because this field will only be shown
+		// if email is invalid (and email will not be shown) so we should show
+		// those errors somewhere. However, only show them if the `alternateEmail`
+		// has not been entered.
+		if (
+			! customErrorMessage &&
+			this.props.contactDetailsErrors?.email &&
+			! this.props.contactDetails?.alternateEmail?.length > 0
```

Comment:
@sirbrillig did you mean:

```
this.props.contactDetails?.alternateEmail?.length === 0
```

or:

```
! this.props.contactDetails?.alternateEmail?.length
```

In the current form, this will luckily work, but only because the integer there gets implicitly converted to a boolean, and that seems to correspond to the expected result here 😉

Reply:
It was a while ago, but I think this was intentional. Here's the possible states:

- `! 'foo@example.com'.length > 0` (evaluates to false, meaning alternate email is set, so we should not show any email field errors)
- `! ''.length > 0` (evaluates to true, meaning alternate email is not set, so we should show any email field errors)
- `! undefined > 0` (evaluates to true, meaning alternate email is not set, so we should show any email field errors)

`this.props.contactDetails?.alternateEmail?.length === 0` would not cover the third case because `undefined === 0` is false.

`! this.props.contactDetails?.alternateEmail?.length` does look like it would cover all the cases. I thought there was some reason I didn't use it here, but at the moment I can't think of why.

Worth noting that a lot of this (certainly awkward) logic should be going away with https://github.com/Automattic/wp-calypso/issues/45336